game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/air/AirAttackVsNonSubsStepTest.java

```diff
@@ -0,0 +1,64 @@
+package games.strategy.triplea.delegate.battle.steps.fire.air;
+
+import static games.strategy.triplea.delegate.battle.MockBattleState.givenBattleState;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnit;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitCanNotBeTargetedBy;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitDestroyer;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitIsAir;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.List;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class AirAttackVsNonSubsStepTest {
+
+  @ParameterizedTest(name = "[{index}] {0} is {2}")
+  @MethodSource
+  void testWhatIsValid(
+      final String displayName, final BattleState battleState, final boolean expected) {
+    final AirAttackVsNonSubsStep underTest = new AirAttackVsNonSubsStep(battleState);
+    assertThat(underTest.valid(), is(expected));
+    if (expected) {
+      assertThat(underTest.getNames(), hasSize(1));
+    } else {
+      assertThat(underTest.getNames(), hasSize(0));
+    }
+  }
+
+  static List<Arguments> testWhatIsValid() {
+    return List.of(
+        Arguments.of(
+            "Attacker has air units and no destroyers vs Defender subs",
+            givenBattleState()
+                .attackingUnits(List.of(givenUnit(), givenUnitIsAir()))
```

Comment:
Nit, most given methods should take a parameter. Something like:
```
givenUnitWithAttackPower(2)
```
If there is no parameter, and the return value is very generic, then something like the following would be typical (note the 'any' word):
```
givenAnyUnit()
```

Otherwise the given should probably spell out what is special about the thing that is being returned, you've done that in the example of: `givenUnitIsAir()`